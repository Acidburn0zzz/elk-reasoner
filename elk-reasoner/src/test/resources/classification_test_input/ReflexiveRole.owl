Prefix( : = <http://example.org/> )
Prefix( owl: = <http://www.w3.org/2002/07/owl#> )
Ontology(
ReflexiveObjectProperty(:R)
ReflexiveObjectProperty(:H)

EquivalentClasses(:B ObjectSomeValuesFrom(:R :A))
EquivalentClasses(:D ObjectSomeValuesFrom(:S :C))
SubObjectPropertyOf(:H :H1)
SubObjectPropertyOf(ObjectPropertyChain(:R :H1) :S) #S is implicitly reflexive since R and H1 are.

#test that :H (an implicitly reflexive role) is properly handled, i.e., :T1 o :T2 -> U
SubObjectPropertyOf(ObjectPropertyChain(:T1 :T2 :H) :U)
SubClassOf(:C1 ObjectSomeValuesFrom(:T1 ObjectSomeValuesFrom(:T2 :C2)))
SubClassOf(ObjectSomeValuesFrom(:U :C2) :F)

#test that :R is properly eliminated from these chains to still entail V1 -> V3 and, thus, K -> L
SubObjectPropertyOf(ObjectPropertyChain(:R :V1) :V2)
SubObjectPropertyOf(ObjectPropertyChain(:V2 :R) :V3)
EquivalentClasses(:K ObjectSomeValuesFrom(:V1 :K1))
EquivalentClasses(:L ObjectSomeValuesFrom(:V3 :K1))
)