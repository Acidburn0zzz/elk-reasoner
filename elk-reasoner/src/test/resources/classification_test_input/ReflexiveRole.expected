Ontology(
Declaration(Class(<http://example.org/C>))
Declaration(Class(<http://example.org/A>))
Declaration(Class(<http://example.org/D>))
Declaration(Class(<http://example.org/B>))
SubClassOf(<http://example.org/A> <http://example.org/B>)
SubClassOf(<http://example.org/C> <http://example.org/D>)
)